src/snowflake/snowpark/modin/plugin/_internal/utils.py

```diff
@@ -166,26 +166,25 @@ def _check_if_sql_query_contains_order_by_and_warn_user(
     """
     # We need to determine if the query contains an ORDER BY, which we can determine by looking at the
     # logical plan. If a query contains an ORDER BY that affects the ordering of the final result,
-    # the logical plan will contain a row whose parent operator is the operation with id 0
-    # (the Result operation), and whose operation is either a Sort (ORDER BY) or a SortWithLimit
-    # (ORDER BY {LIMIT|FETCH}).
-    # Examples of a logical plan with an ORDER BY:
-    #    step   id parentOperators    operation ...
-    # 0   NaN  NaN            None  GlobalStats ...
-    # 1   1.0  0.0            None       Result ...
-    # 2   1.0  1.0             [0]         Sort ...
-    # 3   1.0  2.0             [1]    TableScan ...
+    # the logical plan will contain an operation row whose operator name is Sort.
+    # Example of a logical plan with an ORDER BY:
+    #     id                             operation ...
+    # 0    0                        ResultFinalize ...
+    # 1    1                     Exchange (SINGLE) ...
+    # 2    2                          ResultWorker ...
+    # 3    3                            Projection ...
+    # 4    4                                  Sort ...
+    # 5    5                      Exchange (RANGE) ...
+    # 6    6                        RangePartition ...
+    # 7    7                  Exchange (BROADCAST) ...
+    # 8    8                                Sample ...
+    # 9    9                     Exchange (SINGLE) ...
+    # 10  10                                Sample ...
+    # 11  11                             TableScan ...
+    # 12  12                             TableScan ...
     # (ellipsis denotes additional columns that have been removed since they are unnecessary for this example.)
-    #    step   id parentOperators      operation ...
-    # 0   NaN  NaN            None    GlobalStats ...
-    # 1   1.0  0.0            None         Result ...
-    # 2   1.0  1.0             [0]  SortWithLimit ...
-    # 3   1.0  2.0             [1]      TableScan ...
-    potential_sort_row = logical_plan[
-        (logical_plan["parentOperators"] == "[0]")
-        & (logical_plan["operation"].str.lower().str.contains("sort"))
-    ]
-    contains_order_by = len(potential_sort_row) > 0
+    # Have to use `strip` since the names of the operators contain indentation.
+    contains_order_by = (logical_plan["operation"].str.strip() == "Sort").any()
```

Comment:
Was this is a BCR change or just internal change?

Reply:
Hmm - I'm not actually sure - explain is a public operator, so the output from explain changing could be a BCR?

Reply:
https://snowflake.slack.com/archives/C02BTC3HY/p1714594898428329